writers/ScatterGatherBuffer.go

```diff
@@ -177,3 +177,44 @@ func (sgb *ScatterGatherBuffer) DropFirst(amount uint32) {
 
 	sgb.size -= dropped
 }
+
+func (sgb *ScatterGatherBuffer) TakeFirstUnsafe(n uint32) *ScatterGatherBuffer {
```

Comment:
Don't use words that are too generic. What does "unsafe" even mean in this context?